registry/registry.go

```diff
@@ -478,7 +478,12 @@ func resolveConfiguration(args []string) (*configuration.Configuration, error) {
 
 	config, err := configuration.Parse(fp)
 	if err != nil {
-		return nil, fmt.Errorf("error parsing %s: %v", configurationPath, err)
+		switch err.(type) {
```

Comment:
I think we should leave this as is.

We should still mention the config file in this error since this issue can be triggered by unexpected ENVVAR conflicts, and we're still performing these operations based on the config file, so keeping this would result in a message the following for the linked issue, which seems pretty good:

```
configuration error: error parsing /etc/docker/registry/config.yml: parsing environment variable REGISTRY_LOG_LEVEL: yaml: unmarshal errors...
```

Reply:
I was in two minds about this one, but yes, if the env vars can conflict with the config yaml in such a way that an error happens, I agree!